lib/control-connection.js

```diff
@@ -415,6 +415,26 @@ class ControlConnection extends events.EventEmitter {
     }
   }
 
+  async _refreshControlConnection(hostIterator) {
+
+    try {
+
+      this.connection = this._borrowAConnection(hostIterator);
+    }
+    catch(err) {
```

Comment:
Does the type of error matter here at all?  Is there ever an error where we don't want the retry to happen?